src/tags.rs

```diff
@@ -0,0 +1,188 @@
+mod seals {
+    pub trait TagConversion<T>: Sized {
+        fn __from_inner_type(T) -> Result<Self, T>;
+
+        fn __to_inner_type(&self) -> T;
+    }
+}
+
+pub trait U16Tag: Sized {
```

Comment:
I'm generally critical towards having *public* traits represent common inherent methods. Firstly, it's an evolution risk since they can be named in trait bounds and implemented by library users. Secondly, actually using the methods offered by them requires importing an additional `use` statement of an item which is only remotely connected. Thirdly, the methods are by default collapsed in the documentation and much harder to find (appear in the bottom, would expect mostly standard conversion functions, ..).

In essense, if the trait is not used in a cleaner interface I would rather expose the functionality as plain inherent methods. Feel free to do this via local macro rules, the standard library does it that way and this avoids duplication of code written.

Reply:
These are inherent macro-generated methods now.